mlflow/langchain/langchain_tracer.py

```diff
@@ -165,25 +139,24 @@ def _end_span(
                 status=status,
             )
 
-    def _is_trace_active(self, request_id: str) -> bool:
-        """Check if a trace with the given request ID is active (i.e. not ended yet)"""
-        return (
-            # Case 1: The root span is started by this callback, the ID
-            # should be in the active list, otherwise it's already ended.
-            request_id in self._active_request_ids
-            # Case 2: The root span is created by fluent API outside this callback.
-            # In this case, we check the context to see if the trace is active or not.
-            or (
-                (active_span := mlflow.get_current_active_span())
-                and (active_span.request_id == request_id)
-            )
-            # Case 3: The root span is created by client API outside this callback,
-            # and passed via the `parent_span` argument of the callback. In this case,
-            # we have no way to check if it is active or not, so just assume it is.
-            or self._parent_span
-        )
+        st = self._run_span_mapping.pop(str(run_id), None)
```

Comment:
If above end_span fails and we didn't detach the token, what would happen?

Reply:
The span will not be detached, but will eventually be detached when we call `tracer.flush()` by the end of `_patched_inference`.